### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by dancio**

### CODE REVIEW

File: syncanocli/context.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        self.filename = filename
+        self._data = {}
+        self.read()
+
+    def __getattr__(self, k):
+        if k not in self.__dict__:
+            if k not in self._data:
+                self._data[k] = {}
+
+            section = ConfigSection(self, k)
```
Comment (ilonajulczuk) [inline syncanocli/context.py:80]: The benefit of using ConfigSection isn't clear to me.

**Comment by ilonajulczuk** [comment]
Looks quite nice ;). However, having some tests would be nice. How do you like working with click?

File: syncanocli/context.py (hunk 1 above)
Comment (dancio) [inline syncanocli/context.py:80]: syntax sugar u can access all config properties via objects:(dot notation) e.g: `config.credentials.api_key` this is corresponding to ini files structure e.g `config.{section}.{variable}`

**Comment by dancio** [comment]
I'll add few test in next PR's. `click` is a pretty neat piece of code :) so far it does the job.

File: syncanocli/context.py (hunk 1 above)
Comment (ilonajulczuk) [inline syncanocli/context.py:80]: :+1:

**Pull request merged**